src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/CollectF1R2Counts.java

```diff
@@ -0,0 +1,233 @@
+                }
+            }
+        });
+
+        // Intentionally not use try-with-resources so that the writer stays open outside of the try block
+        try {
+            altTableWriter = new AltSiteRecordTableWriter(altDataTable);
+        } catch (IOException e) {
+            throw new UserException(String.format("Encountered an IO exception creating a writer for %s", altDataTable), e);
+        }
+
+    }
+
+    @Override
+    public void apply(final AlignmentContext alignmentContext, final ReferenceContext referenceContext, final FeatureContext featureContext) {
+        final int position = referenceContext.getInterval().getStart();
+        final String refContext = referenceContext.getKmerAround(position, F1R2FilterConstants.REF_CONTEXT_PADDING);
+        // final String refContext = Arrays.toString(referenceContext.getKmerAround(position, F1R2FilterConstants.REF_CONTEXT_PADDING));
+        final Nucleotide refBase = F1R2FilterUtils.getMiddleBase(refContext);
+
+        // START deprecated code
+        referenceContext.setWindow(F1R2FilterConstants.REF_CONTEXT_PADDING, F1R2FilterConstants.REF_CONTEXT_PADDING);
+        final String oldRefContext = new String(referenceContext.getBases());
+        Utils.validate(refContext.equals(oldRefContext), String.format("old ref context %s and new ref context %s disagree", oldRefContext, refContext));
+        // END deprecated code
+
+        if (refContext.contains("N") || refContext.length() != F1R2FilterConstants.REFERENCE_CONTEXT_SIZE) {
+            return;
+        }
+
+        if (refContext == null) {
+            logger.warn(String.format("Skipped a site with null reference at interval %s, k-mer = %s",
+                    referenceContext.getInterval().toString(), refContext));
+            return;
+        }
+
+        final ReadPileup pileup = alignmentContext.getBasePileup().makeFilteredPileup(pe -> pe.getQual() > MINIMUM_BASE_QUALITY);
+        final int[] baseCounts = pileup.getBaseCounts();
+        final int depth = (int) MathUtils.sum(baseCounts);
+
+        if (!isPileupGood(pileup, referenceContext)) {
+            return;
+        }
+
+        // Make a copy of base counts and update the counts of ref to -1. Now the maxElementIndex of the array gives us
+        // the alt base.
+        final int[] baseCountsCopy = Arrays.copyOf(baseCounts, baseCounts.length);
+        baseCountsCopy[refBase.ordinal()] = -1;
+        final int altBaseIndex = MathUtils.maxElementIndex(baseCountsCopy);
+        final boolean referenceSite = baseCounts[altBaseIndex] == 0;
+
+        // If the site is ref, we simply update the coverage histogram
+        if (referenceSite) {
+            refSiteHistograms.get(refContext).increment(depth <= F1R2FilterConstants.maxDepth ?
+                    depth : F1R2FilterConstants.maxDepth);
+            return;
+        }
+
+        // If we got here, we have an alt site with a single alt base
```

Comment:
Is this comment true?

Reply:
Deleted "with a single alt base." The code ignores other alt alleles when there is more than one alt.